### PR TITLE
Refactor DD4hep/Acts integration

### PR DESCRIPTION
**Pull request by rahmans1**
### Briefly, what does this PR introduce?
[Refactor the DD4hep/Acts integration for acts@20.0.0: and dd4hep@1.21:](https://github.com/eic/epic/issues/34#top)

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #34)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Yes. Transition from ACTS 19 to 20 is backwards incompatible. All xml files and plugins related to trackers need to be updated. Details are provided in the issue. 

### Does this PR change default behavior?
No

### CODE REVIEW

File: src/TrapEndcapTracker_geo.cpp [hunk 1]
```diff
@@ -58,24 +50,19 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
...
+
+  // Add the volume boundary material if configured
+  for (xml_coll_t bmat(x_det, _Unicode(boundary_material)); bmat; ++bmat) {
+    xml_comp_t x_boundary_material = bmat;
+    DD4hepDetectorHelper::xmlToProtoSurfaceMaterial(x_boundary_material, params,
+                                         "boundary_material");
   }
 
   assembly.setVisAttributes(description.invisible());
-  sens.setType("tracker");
```
Comment (wdconinc) [inline src/TrapEndcapTracker_geo.cpp:78]: Are we sure we should remove this? This is unrelated to Acts.

File: configurations/grandteton.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  fields:
+    marco:
+  beampipe:
+  tracking:
+    silicon_disks:
+    silicon_barrel:
+    vertex_barrel:
+    mpgd_barrel:
+    tof_barrel:
+    tof_endcap:
```
Comment (wdconinc) [inline configurations/grandteton.yml:11]: `git rm configurations/grandteton.yml`

File: src/B0Preshower_geo.cpp [hunk 3]
```diff
@@ -210,9 +192,7 @@ static Ref_t create_B0Preshower(Detector& description, xml_h e, SensitiveDetecto
             PlacedVolume sens_pv = sensVols[ic];
             DetElement   comp_elt(r_module, sens_pv.volume().name(), mod_num);
             comp_elt.setPlacement(sens_pv);
-            // std::cout << " adding ACTS extension" << "\n";
-            Acts::ActsExtension* moduleExtension = new Acts::ActsExtension("XZY");
-            comp_elt.addExtension<Acts::ActsExtension>(moduleExtension);
```
Comment (wdconinc) [inline src/B0Preshower_geo.cpp:215]: Do we need this here?
```
          auto &params = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(comp_elt);
          params.set("axis_definitions", "XZY");
```

File: src/B0Preshower_geo.cpp [hunk 4]
```diff
@@ -196,9 +180,7 @@ static Ref_t create_B0Preshower(Detector& description, xml_h e, SensitiveDetecto
             PlacedVolume sens_pv = sensVols[ic];
             DetElement   comp_elt(module, sens_pv.volume().name(), mod_num);
             comp_elt.setPlacement(sens_pv);
-            // std::cout << " adding ACTS extension" << "\n";
-            Acts::ActsExtension* moduleExtension = new Acts::ActsExtension("XZY");
-            comp_elt.addExtension<Acts::ActsExtension>(moduleExtension);
```
Comment (wdconinc) [inline src/B0Preshower_geo.cpp:201]: Do we need this here?
```
          auto &params = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(comp_elt);
          params.set("axis_definitions", "XZY");
```

File: src/EndcapTOF_geo.cpp [hunk 5]
```diff
@@ -184,13 +183,12 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
...
-  detlayer->addValue(10. * mm, "z_min", "envelope");
-  detlayer->addValue(10. * mm, "z_max", "envelope");
-  detlayer->addType("sensitive plane", "layer");
-  layer_detEl.addExtension<Acts::ActsExtension>(detlayer);
+  auto &layerParams = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(
+                      layer_detEl);
+  layerParams.set<double>("envelope_r_min", -80. * mm);
+  layerParams.set<double>("envelope_r_max", 670. * mm);
+  layerParams.set<double>("envelope_z_min", 10. * mm);
+  layerParams.set<double>("envelope_z_max", 10. * mm);
```
Comment (wdconinc) [inline src/EndcapTOF_geo.cpp:191]: Can/should we put this into the compact file?

File: src/B0Preshower_geo.cpp (hunk 3 above)
Comment (rahmans1) [reply src/B0Preshower_geo.cpp:215]: Nothing in B0 is integrated with the main tracking. Will deal with these with a later PR.

File: configurations/grandteton.yml (hunk 2 above)
Comment (rahmans1) [reply configurations/grandteton.yml]: Done

File: src/B0Preshower_geo.cpp (hunk 3 above)
Comment (rahmans1) [reply src/B0Preshower_geo.cpp:215]: Also preshower I think is deprecated and replaced by the B0 ECal. Moving it to deprecated folder.

Comment (wdconinc) [reply src/B0Preshower_geo.cpp:215]: Ok, I'm just pointing this out since it removes code but doesn't replace it with equivalent code. It may lead to difficulties in the future when someone adapts this code, and isn't aware of the past "XZY" axis definitions. But I leave it up to you to decide how to deal with this.

Comment (rahmans1) [reply src/B0Preshower_geo.cpp:215]: Btw is there a deprecated folder anymore? Seems like everything is in the same space. Would comment at the top of the constructor like "Deprecated. ACTS tracking interface not functional." do?

Comment (wdconinc) [reply src/B0Preshower_geo.cpp:215]: Sure, I think that's probably sufficient.

File: src/EndcapTOF_geo.cpp (hunk 5 above)
Comment (rahmans1) [reply src/EndcapTOF_geo.cpp]: I have for now put this [here](https://github.com/eic/epic/blob/34-refactor-the-dd4hep-acts-integration-for-acts-20-0-0-and-dd4hep-1-21/compact/tracking/tof_endcap.xml#L146) and parsed [here](https://github.com/eic/epic/blob/34-refactor-the-dd4hep-acts-integration-for-acts-20-0-0-and-dd4hep-1-21/src/EndcapTOF_geo.cpp#L54). But this is not ideal. The intended parametrization is not clear to me. I will ask nicholas how they were derived. There is also another envelope tag defined in the compact file which is never parsed. Should sort this in a separate PR like the MPGD.

File: src/TrapEndcapTracker_geo.cpp (hunk 1 above)
Comment (rahmans1) [reply src/TrapEndcapTracker_geo.cpp]: Moved it to the top for clarity. Only needs to be invoked once.

**Review by wdconinc — APPROVED** [review]
